### PR TITLE
Move RabbitMQ to a dedicated conda environment.

### LINKED ISSUE #39: rabbitmq version for aiida 2.0 in aiida docker image
ubuntu 20.04 ships with rabbitmq 3.8.2 which leads to

```
Warning: RabbitMQ v3.8.2 is not supported and will cause unexpected problems!
Warning: It can cause long-running workflows to crash and jobs to be submitted multiple times.
Warning: See https://github.com/aiidateam/aiida-core/wiki/RabbitMQ-version-to-use for details.
``` 

So, we now need to decide whether to use an old version of rabbitmq or a new version with proper configuration.

I'm leaning towards the latter solution (otherwise we're stuck with an old version for which there is no maintenance).

I think the [corresponding section of the wiki](https://github.com/aiidateam/aiida-core/wiki/RabbitMQ-version-to-use#using-a-recent-rabbitmq-version) should mention (1) which configuration values to change, and whether (2) one still needs to remove some temporary files or not.

It should also include instructions on how to disable the warning once rabbitmq has been configured correctly.

Thoughts @chrisjsewell @giovannipizzi @csadorf ?

### PR DESCRIPTION
**Pull request by yakutovicha**
fixes #39

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -47,14 +47,14 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-ins
 RUN apt-get update && apt-get install -y --no-install-recommends  \
     build-essential       \
     bzip2                 \
+    erlang                \
```
Comment (unkcpz) [inline Dockerfile:50]: Why do we need this? I think conda for rabbitmq will take care of the correct version of erlang and apt install of rabbitmq-server for arm64 will install this with required version https://www.rabbitmq.com/install-debian.html#apt-quick-start-cloudsmith.

File: Dockerfile [hunk 2]
```diff
@@ -95,14 +95,29 @@ RUN cd /tmp && \
...
 # Copy the script load-singlesshagent.sh to /usr/local/bin.
 COPY bin/load-singlesshagent.sh /usr/local/bin/load-singlesshagent.sh
 
 # Create system user.
 COPY my_init.d/create-system-user.sh /etc/my_init.d/10_create-system-user.sh
 
 # Launch rabbitmq server
-COPY my_init.d/start-rabbitmq-${TARGETARCH}.sh /etc/my_init.d/20_start-rabbitmq.sh
+COPY opt/start-rabbitmq-${TARGETARCH}.sh /opt/start-rabbitmq.sh
+COPY my_init.d/start-rabbitmq.sh /etc/my_init.d/20_start-rabbitmq.sh
```
Comment (unkcpz) [inline Dockerfile:120]: I am confused with these. 

Ah, I see. The `/opt/start-rabbitmq.sh` will be different for arm64 and amd64, and start-rabbitmq.sh will used for both?

**Review by unkcpz — COMMENTED** [review]
@yakutovicha thanks! Looks good to me and not sure why test failed. Seems like the rabbitmq from conda not properly started. 

Instead of blocking you moving forward to migrate to aiida 2.0, I'll approve this once CI test passes and test it for arm64 on my computer next week.

File: Dockerfile (hunk 1 above)
Comment (yakutovicha) [reply Dockerfile:50]: Yes, that is what I was hoping for. But it didn't work, unfortunately :(. I will create an issue about that hack.

File: Dockerfile (hunk 2 above)
Comment (yakutovicha) [reply Dockerfile:120]: > Ah, I see. The `/opt/start-rabbitmq.sh` will be different for arm64 and amd64, and start-rabbitmq.sh will used for both?

yep!

**Comment by yakutovicha** [comment]
> Looks good to me and not sure why test failed.

`my_init.d/start-rabbitmq.sh` simply didn't have execution rights, see 9364c62

**Review by unkcpz — APPROVED** [review]
Thanks!

File: Dockerfile (hunk 1 above)
Comment (yakutovicha) [reply Dockerfile:50]: here is the issue: https://github.com/aiidateam/aiida-prerequisites/issues/58

**Pull request merged**